src/drivers/github.js

```diff
@@ -713,6 +722,76 @@ class Github {
   get userName() {
     return 'GitHub Action';
   }
+
+  async checkWorkflowForTimeoutMinutes() {
+    winston.warn(
+      'Github Actions timeout has been updated from 72h to 35 days. Update your workflow accordingly to be able to restart it automatically.'
+    );
+    const WARNINGS = [];
+    const _DIR = '.github/workflows/';
+    const yaml = require('js-yaml');
+    const files = await fs.readdir(_DIR);
+    for (const file of files) {
+      const rawStr = await fs.readFile(`${_DIR}${file}`, 'utf8');
+      const doc = yaml.load(rawStr);
+      Object.keys(doc.jobs).forEach((job) => {
+        // does job contain timeout?
+        const timeoutVal = doc.jobs[job]['timeout-minutes'];
+        if (timeoutVal === undefined) return;
+        if (parseInt(timeoutVal) === 50400) return;
+        // does job have label runs-on "self-hosted" or "cml"
+        const runsOn = doc.jobs[job]['runs-on'];
+        switch (typeof runsOn) {
+          case 'string':
+            if (
+              runsOn !== 'self-hosted' &&
+              !runsOn.toLocaleLowerCase().includes('cml')
```

Comment:
Users are not enforced to set `cml`

Reply:
Correct, this is an attempt to reduce false positives; I'm airing on the side of being less intrusive on the warnings.

Reply:
I think that warning if they set that specific amount of time is a pretty sure asumption

Reply:
if by the specific amount you mean: `timeout-minutes: 4320` I can agree with that argument, but I was explicitly trying to avoid warning simply on the presence of `timeout-minutes` since there are plenty of valid reasons for that to be used.

Reply:
ok